dashboard/app/serializers/api/v1/section_instructor_serializer.rb

```diff
@@ -36,6 +36,10 @@ def section_name
   end
 
   def instructor_name
-    object.instructor.name
+    object.instructor&.name if object.active?
```

Comment:
This does change other endpoints, but is similar to the `section_instructor_info_serializer` that section summaries use. We don't want to return the name unless the teacher has accepted the invitation.